cli_tools/gce_ovf_import/ovf_importer/ovf_importer.go

```diff
@@ -460,17 +405,47 @@ func (oi *OVFImporter) CleanUp() {
 	}
 }
 
-func (oi *OVFImporter) importWithModule(osID string) error {
+func (oi *OVFImporter) importWithModule(osID string, diskInfos *[]ovfutils.DiskInfo) error {
 	var dataDiskURIs []string
-	for _, info := range *oi.diskInfos {
+	for _, info := range *diskInfos {
 		dataDiskURIs = append(dataDiskURIs, info.FilePath)
 	}
 	params := *oi.params
 	params.OsID = osID
 	params.Deadline = params.Deadline.Add(-1 * instanceConstructionTime)
-	imageURIs, err := oi.multiImageImporter.Import(oi.ctx, oi.params, dataDiskURIs)
+	imageURIs, err := oi.multiImageImporter.Import(oi.ctx, &params, dataDiskURIs)
 	if err == nil {
 		oi.imageURIs = imageURIs
 	}
 	return err
 }
+
+// getOsIDValue determines the osID to use when importing the boot disk.
+func (oi *OVFImporter) getOsIDValue(descriptor *ovf.Envelope) (osIDValue string, err error) {
+	userOS := oi.params.OsID
+	descriptorOS, err := ovfutils.GetOSId(descriptor)
+	if err != nil {
+		oi.Logger.Debug(fmt.Sprintf("Didn't find valid osID in descriptor. Error=%q", err))
+		descriptorOS = ""
+	}
+	oi.Logger.Debug(fmt.Sprintf("osID candidates: from-user=%q, ovf-descriptor=%q", userOS, descriptorOS))
+
+	if userOS != "" {
+		osIDValue = userOS
+	} else if descriptorOS != "" {
+		osIDValue = descriptorOS
+	} else {
+		oi.Logger.User("OS will be detected from boot disk.")
+		return "", nil
+	}
+
```

Comment:
Consider printing a warning that the OS in the descriptor differs from the OS provided by user

Reply:
done!